test/e2e/specs/editor/various/block-bindings.spec.js

```diff
@@ -0,0 +1,827 @@
+						.getByRole( 'button', { name: 'Unlink' } )
+				).toBeHidden();
+			} );
+
+			test( 'Should lock url and text controls when both are bound', async ( {
+				editor,
+				page,
+			} ) => {
+				await editor.insertBlock( multipleBindingsButtonBlock );
+				const buttonBlock = editor.canvas.getByRole( 'document', {
+					name: 'Block: Button',
+					exact: true,
+				} );
+				await buttonBlock.click();
+
+				// Alignment controls are visible.
+				await expect(
+					page.getByRole( 'button', {
+						name: alignName,
+					} )
+				).toBeVisible();
+
+				// Format controls don't exist.
+				await expect(
+					page.getByRole( 'button', {
+						name: boldName,
+					} )
+				).toBeHidden();
+
+				// Button is not editable.
+				const isContentEditable = await buttonBlock
+					.locator( 'div' )
+					.getAttribute( 'contenteditable' );
+				expect( isContentEditable ).toBe( 'false' );
+
+				// Link controls don't exist.
+				await expect(
+					page
+						.getByRole( 'toolbar', { name: 'Block tools' } )
+						.getByRole( 'button', { name: 'Link' } )
+				).toBeHidden();
+				await expect(
+					page
+						.getByRole( 'toolbar', { name: 'Block tools' } )
+						.getByRole( 'button', { name: 'Unlink' } )
+				).toBeHidden();
+			} );
+		} );
+
+		test.describe( 'Image', () => {
+			test( 'Should show the upload form when url is not bound', async ( {
+				editor,
+			} ) => {
+				await editor.insertBlock( { name: 'core/image' } );
+				const imageBlock = editor.canvas.getByRole( 'document', {
+					name: 'Block: Image',
+				} );
+				await imageBlock.click();
+				await expect(
+					imageBlock.getByRole( 'button', { name: 'Upload' } )
+				).toBeVisible();
+			} );
+
+			test( 'Should NOT show the upload form when url is bound', async ( {
+				editor,
+			} ) => {
+				await editor.insertBlock( urlBindingImageBlock );
+				const imageBlock = editor.canvas.getByRole( 'document', {
+					name: 'Block: Image',
+				} );
+				await imageBlock.click();
+				await expect(
+					imageBlock.getByRole( 'button', { name: 'Upload' } )
+				).toBeHidden();
+			} );
+
+			test( 'Should lock url controls when url is bound', async ( {
+				editor,
+				page,
+			} ) => {
+				await editor.insertBlock( urlBindingImageBlock );
+				const imageBlock = editor.canvas.getByRole( 'document', {
+					name: 'Block: Image',
+				} );
+				await imageBlock.click();
+
+				// Replace controls don't exist.
+				await expect(
+					page.getByRole( 'button', {
+						name: imageReplaceName,
+					} )
+				).toBeHidden();
+
+				// Image placeholder doesn't show the upload button.
+				await expect(
+					imageBlock.getByRole( 'button', { name: 'Upload' } )
+				).toBeHidden();
+
+				// Alt textarea is enabled and with the original value.
+				await expect( page.getByLabel( imageAltLabel ) ).toBeEnabled();
+				const altValue = await page
+					.getByLabel( imageAltLabel )
+					.inputValue();
+				expect( altValue ).toBe( 'a' );
```

Comment:
What's the reason it has an alt value set to `a`? The same applies to the title?

Reply:
I'm adding a default value in the block declaration here: [link](https://github.com/WordPress/gutenberg/pull/58550/files#diff-9bd7430ffd5302c7b9b110debae97bde7c68bf34f9e3908c34959c42397c9f26R135-R136). The idea is to check that it is not overridden at any point.

Reply:
I would prefer something more descriptive, like `default alt value`.

Reply:
Done in [this commit](https://github.com/WordPress/gutenberg/pull/58550/commits/2961b457d7fa9cce1d4cdef9247f365251aa54b6).